src/tiledb/cloud/soma/ingest.py

```diff
@@ -155,15 +240,21 @@ def build_ingest_workflow_graph(
                 extra_tiledb_config=extra_tiledb_config,
                 ingest_mode=ingest_mode,
                 platform_config=platform_config,
-                resources=_DEFAULT_RESOURCES if resources is None else resources,
-                access_credentials_name=access_credentials_name,
+                resources=carry_along["resources"],
+                access_credentials_name=carry_along["access_credentials_name"],
+                logging_level=logging_level,
                 dry_run=dry_run,
             )
             collector.depends_on(node)
 
-        return grf
+    else:
+        raise ValueError(f"input_uri {input_uri!r} is neither file nor directory")
+
+    grf.compute()
+    grf.wait()
```

Comment:
Is waiting on this the desired outcome? It seems like you would want to kick it off, then return the graph ID once you’ve done so.

Reply:
Sorry @thetorpedodog I'm not following the verbal

Can you please propose a code mod?

Reply:
@thetorpedodog ping 🙏

Reply:
Just remove the `grf.wait()` here, since after you call `.compute` the actual ingestion graph has already started. With the `.wait` here, the graph-starting function waits for the actual ingestion graph to complete, which seems like not what you want.